tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc

```diff
@@ -0,0 +1,354 @@
+    
+    // Run and Check for Cooling Mode
+    Mode = 0;
+    Temp = 15;   	
+    ZoneIndex = 1; 
+    
+    ZoneSysEnergyDemand( ZoneIndex ).OutputRequiredToCoolingSP = -2716.6229;
+    ZoneSysEnergyDemand( ZoneIndex ).OutputRequiredToHeatingSP = -45507.8487;
+
+    DXCoil( CoolCoilIndex ).RatedAirMassFlowRate( 1 ) = 0.2066;
+    DXCoil( CoolCoilIndex ).InletAirTemp = 25.5553;
+    DXCoil( CoolCoilIndex ).InletAirHumRat = 8.4682e-3;
+    DXCoil( CoolCoilIndex ).InletAirEnthalpy = 47259.78;
+    
+    CalcVRFIUAirFlow(ZoneIndex, Mode, Temp, CoolCoilIndex, HeatCoilIndex, SHSCModify, FanSpdRatio, Wout, Toutlet, Houtlet, HcoilIn, TcIn, SHact, SCact);
+    EXPECT_NEAR( TcIn, 25.56, 0.01 );
+    EXPECT_NEAR( Toutlet, 17.89, 0.01 );
+    EXPECT_NEAR( Houtlet, 39440, 1 );
+    EXPECT_NEAR( HcoilIn, 47259, 1 );
+    EXPECT_NEAR( SHact, 3.00, 0.01 );
+    
+    
+    // Run and Check for Heating Mode
+    Mode = 1;
+    Temp = 42;   	
+    ZoneIndex = 2; 
+    
+    ZoneSysEnergyDemand( ZoneIndex ).OutputRequiredToCoolingSP = 43167.2628;
+    ZoneSysEnergyDemand( ZoneIndex ).OutputRequiredToHeatingSP = 4241.66099;
+
+    DXCoil( HeatCoilIndex ).RatedAirMassFlowRate( 1 ) = 0.21136;
+    DXCoil( HeatCoilIndex ).InletAirTemp = 20.2362;
+    DXCoil( HeatCoilIndex ).InletAirHumRat = 4.1053e-3;
+    DXCoil( HeatCoilIndex ).InletAirEnthalpy = 30755.6253;
+    
+    CalcVRFIUAirFlow(ZoneIndex, Mode, Temp, CoolCoilIndex, HeatCoilIndex, SHSCModify, FanSpdRatio, Wout, Toutlet, Houtlet, HcoilIn, TcIn, SHact, SCact);
+    EXPECT_NEAR( TcIn, 20.24, 0.01 );
+    EXPECT_NEAR( Toutlet, 38.37, 0.01 );
+    EXPECT_NEAR( Houtlet, 49113, 1 );
+    EXPECT_NEAR( HcoilIn, 30756, 1 );
+    EXPECT_NEAR( SCact, 5.00, 0.01 );
+    
+    // Clean up
+    DXCoil.deallocate( ); 
+    ZoneSysEnergyDemand.deallocate( ); 
+}
+
+TEST( HVACVariableRefrigerantFlow, VRF_FluidTCtrl_CalcVRFIUTeTc )
+{
+    // PURPOSE OF THIS TEST:
+    //   Test the method CalcVRFIUTeTc_FluidTCtrl, which determines the VRF evaporating temperature at 
+    //   cooling mode and the condensing temperature at heating mode.
+    	
+    using namespace HVACVariableRefrigerantFlow;
+    
+    // Allocate
+    int NumVRFCondenser = 1;
+    VRF.allocate( NumVRFCondenser );
+    int NumTUList = 1;
+    TerminalUnitList.allocate( NumTUList );
+    
+    // Common Inputs
+    int IndexVRFCondenser = 1;
+    int IndexTUList = 1;
+    
+    TerminalUnitList( IndexTUList ).NumTUInList = 2;
+    
+    VRF( IndexVRFCondenser ).ZoneTUListPtr = 1;
+    VRF( IndexVRFCondenser ).AlgorithmIUCtrl = 0;
+    VRF( IndexVRFCondenser ).EvapTempFixed = 3;
+    VRF( IndexVRFCondenser ).CondTempFixed = 5;
+    
+    // Run and Check 
+    CalcVRFIUTeTc_FluidTCtrl( IndexVRFCondenser );
+    
+    EXPECT_EQ( VRF( IndexVRFCondenser ).IUEvaporatingTemp, 3 );
+    EXPECT_EQ( VRF( IndexVRFCondenser ).IUCondensingTemp, 5 );
+    
+    // Clean up
+    VRF.deallocate( ); 
+    TerminalUnitList.deallocate( ); 
+}
+
+}
```

Comment:
Hi @rraustad ,

Thanks for the careful review and insightful comments! Really appreciate that! I will update the codes to address your comments soon.

(1) The chart showing the code hierarchy can be found via
https://www.dropbox.com/s/5xd05e49rx6r03b/VRF-HP%20Coding%20Hierarhcy_zrp.PNG?dl=0
(2) I have checked the 4 files that created diffs. I run them with the develop branch codes and my codes, and found the simulation results (eso file) exactly the same. I think the difference comes from the new added output variables. I will have further look at this. Details can be found via https://www.dropbox.com/sh/lnffseq0vws1h2n/AAAZzQLmeORwdJRRT0nSXA65a?dl=0

Best,
Rongpeng

@hongtz68